### PR TITLE
This patch fixes a bug where a redirect causes fetched pages to be in…

### PR DESCRIPTION
**Pull request by npe9**
…terpreted as utf8, breaking icon download.

I creates a new method fetchBytes in UtilHttpClient and makes fetchPage a special case of fetchBytes. This centralizes the http client machinery to UtilHttpClient and all calls to http servers should be made through the class rather than relying on other mechanisms.

### CODE REVIEW

File: ffb-client-logic/pom.xml
Comment (christerk) [inline ffb-client-logic/pom.xml:1]: This pom update does not seem to be related to this change. Is there a reason for this dependency being added?

File: ffb-client-logic/src/main/java/com/fumbbl/ffb/client/IconCache.java [hunk 1]
```diff
@@ -269,7 +264,7 @@ public void loadIconFromUrl(String pUrl) {
 			URL iconUrl = null;
 			try {
 				iconUrl = new URL(pUrl);
-				BufferedImage icon = ImageIO.read(iconUrl);
+				BufferedImage icon = ImageIO.read(new ByteArrayInputStream(UtilHttpClient.fetchBytes(pUrl)));
```
Comment (christerk) [inline ffb-client-logic/src/main/java/com/fumbbl/ffb/client/IconCache.java:267]: The created stream is never closed, and will cause a memory leak. You should be using a try block to ensure that the stream is closed. Basic example of this: https://www.infoq.com/news/2010/08/arm-blocks/

In addition, this is impacted by the comment on UtilHttpClient, so refer to that comment as well.

**Review by christerk — CHANGES_REQUESTED** [review]
Overall a good suggestion for how to fix an underlying problem, but there are some details that need to be addressed. Some minor style issues, but also a slightly more serious memory leak, and an unnecessary byte array being created.

**Review by npe9 — COMMENTED** [review]
added close()

File: ffb-client-logic/pom.xml
Comment (npe9) [reply ffb-client-logic/pom.xml:1]: The pom update is to allow the ffb-tools HttpClient to be used instead of ImageIOs native http.

File: ffb-client-logic/src/main/java/com/fumbbl/ffb/client/IconCache.java (hunk 1 above)
Comment (npe9) [reply ffb-client-logic/src/main/java/com/fumbbl/ffb/client/IconCache.java]: I believe this is resolved by adding a close. Please tell me if I forgot anything.

File: ffb-client-logic/pom.xml
Comment (SanityResort) [reply ffb-client-logic/pom.xml:1]: Adding that dependency has a few implications as the tools jar will then have to be added to the dependencies to be doenloaded by each user.

That increases traffic and also waiting time. Plus christer needs to adjust the jnlp file.

I would suggest moving the class to the commons module and have the tools module depend on that.

Also please use the artifactid property for internal dependencies so they always stay up to date as the project moves forward

Comment (npe9) [reply ffb-client-logic/pom.xml:1]: Gotcha, so just to be clear, you propose moving the httpclient modules from tools to commons?

Comment (SanityResort) [reply ffb-client-logic/pom.xml:1]: I did some poking etc. and think we should use the current apache http client. I created a branch `http-client` with a basic implementation that seems to work for downloading icons.

Can you check out that branch and see if it works for the replays you have issues with? If it works the client should also be used when loading pitches from urls.

If we get everything to work we can then decide if we add the lib and it's dependencies as regular dependencies to the client download or if we shade them into the logic artifact

**Comment by SanityResort** [comment]
Icon loading is now done via http client and seems to resolve the icon issue for me.
